R/cre.R

```diff
@@ -20,7 +20,77 @@
 #'
 #' @return a list containing a select list of causal rules, Conditional Average Treatment Effect estimates, and a sensitivity analysis
 #'
-cre <- function(y, z, X, ratio_dis, ite_method_dis, ite_method_inf, ntrees, min_nodes,
-                max_nodes, rules_method, t){
-  #TBD
+cre <- function(y, z, X, ratio_dis, ite_method_dis, ite_method_inf, ntrees, min_nodes, max_nodes, rules_method, t) {
+  # Load Libraries
+  lapply(c("tidyverse", "BART", "bcf", "randomForest", "xgboost", "gbm", "inTrees", "stabs", "rpart", "grf", "glmnet"),
+         library, character.only = TRUE)
+
+  # Step 1: Split data
+  print("Step 1: Splitting Data")
+  if (!("matrix" %in% class(X))) {
+    X <- as.matrix(X)
+  }
+  subgroups <- split_data(y, z, X, ratio_dis)
+  discovery <- subgroups[[1]]
+  inference <- subgroups[[2]]
+
+  # Generate y, z, and X for discovery and inference data
+  y_dis <- discovery[,"y"]
+  z_dis <- discovery[,"z"]
+  X_dis <- discovery[,c("x1", "x2", "x3", "x4", "x5", "x6", "x7", "x8", "x9", "x10")]
```

Comment:
Ah yes, good point. Since we're ensuring in `split_data` that the order of columns is y, z, X, I can recode this using 1 for "y", 2 for "z", and 3:ncol(discovery) as "X".